### PR TITLE
dev: use goreleaser to create Docker images

### LINKED ISSUE #4368: Version missing from golangci-lint in Docker image since v1.56.0
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've read the typecheck section of the FAQ (https://golangci-lint.run/usage/faq/#why-do-you-have-typecheck-errors).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

Since v1.56.0, the version component when running `golangci-lint` in the official Docker images is missing.

I believe https://github.com/golangci/golangci-lint/pull/4279 is connected with the missing version.

### Version of golangci-lint

<details>
N/A
</details>

### Configuration

<details>
N/A
</details>

### Go environment

<details>
N/A
</details>

### Verbose output of running

<details>
N/A
</details>

### A minimal reproducible example or link to a public repository

```sh
$ docker run --rm -ti golangci/golangci-lint:v1.55.2-alpine golangci-lint version
golangci-lint has version v1.55.2 built with go1.21.3 from e3c2265f on 2023-11-03T13:27:48Z
```

vs.

```sh
$ docker run --rm -ti golangci/golangci-lint:v1.56.0-alpine golangci-lint version
golangci-lint has version  built with go1.22.0 from 1a00b409 on 2024-02-07T14:52:31Z
```

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

### PR DESCRIPTION
**Pull request by ldez**
Ends with the problem of the usage of different binaries between Docker images and the pre-compiled binaries.

<details>

#### build

```bash
goreleaser release --skip=publish --skip=validate --clean --snapshot
```

To build quickly, I recommend modifying the `.goreleaser.yml` file:
```yml
#...

builds:
  - binary: golangci-lint
    goos:
#      - darwin
#      - windows
      - linux
#      - freebsd
#      - netbsd
#      - illumos
    goarch:
      - amd64
      - arm64
#      - arm
#      - 386
#      - ppc64le
#      - s390x
#      - mips64
#      - mips64le
#      - riscv64
#      - loong64

#...
```

#### run

```bash
docker run --rm golangci/golangci-lint:latest-amd64 golangci-lint version
```
or if you are on linux/arm64
```bash
docker run --rm golangci/golangci-lint:latest-arm64 golangci-lint version
```

#### labels

```bash
docker inspect golangci/golangci-lint:latest-amd64
```
or if you are on linux/arm64

```bash
docker inspect golangci/golangci-lint:latest-arm64
```

#### More

If you want an "in production" example, I already did that on several projects, you can show this example:
- https://github.com/go-acme/lego/blob/master/.goreleaser.yml
- https://github.com/go-acme/lego/blob/master/.github/workflows/release.yml
- https://hub.docker.com/r/goacme/lego/tags

</details>

Note: images will be available with the same tags as before, the PR just adds new tags and arch.
https://hub.docker.com/r/golangci/golangci-lint/tags

Fixes #4368
Closes #4369

### CODE REVIEW

File: .github/workflows/tag.yml [hunk 1]
```diff
@@ -81,18 +29,10 @@ jobs:
       - name: Login do docker.io
         run: docker login -u golangci -p ${{ secrets.GOLANGCI_LINT_DOCKER_TOKEN }}
 
-      - name: Build and publish ${{ matrix.target.Dockerfile }}
-        uses: docker/build-push-action@v5
+      - name: Create release
+        uses: goreleaser/goreleaser-action@v4
```
Comment (Antonboom) [inline .github/workflows/tag.yml:33]: before we used `v5`, what the reason of downgrade?

File: build/Dockerfile [hunk 2]
```diff
@@ -1,22 +0,0 @@
-# stage 1 building the code
-FROM golang:1.22 as builder
-
-ARG VERSION
-ARG SHORT_COMMIT
-ARG DATE
-
-COPY / /golangci
-WORKDIR /golangci
-RUN APP_VERSION=${VERSION#v} \
```
Comment (Antonboom) [inline build/Dockerfile:10]: I don't see the passing of these variables (`version`, `commit`, `date`) in the new approach.
How does `golangci-lint version` work now?

Comment (Antonboom) [reply build/Dockerfile:10]: UPD: I saw in `.goreleaser.yml`

File: .goreleaser.yml [hunk 3]
```diff
@@ -132,3 +132,118 @@ nfpms:
...
+      - 'golangci/golangci-lint:{{ .Tag }}-arm64'
+  - name_template: 'golangci/golangci-lint:v{{ .Major }}.{{ .Minor }}'
+    image_templates:
+      - 'golangci/golangci-lint:v{{ .Major }}.{{ .Minor }}-amd64'
+      - 'golangci/golangci-lint:v{{ .Major }}.{{ .Minor }}-arm64'
+  - name_template: 'golangci/golangci-lint:{{ .Tag }}'
+    image_templates:
+      - 'golangci/golangci-lint:{{ .Tag }}-amd64'
+      - 'golangci/golangci-lint:{{ .Tag }}-arm64'
+      - 'golangci/golangci-lint:{{ .Tag }}-armv7'
```
Comment (Antonboom) [inline .goreleaser.yml:150]: `armv7` unused?

**Review by Antonboom — APPROVED** [review]

**Review by bombsimon — APPROVED** [review]
LGTM

**Pull request merged**